test/langtools/jdk/javadoc/doclet/testSeeLinkAnchor/TestSeeLinkAnchor.java

```diff
@@ -0,0 +1,300 @@
+ * @run main TestSeeLinkAnchor
+ */
+
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import builder.ClassBuilder;
+import builder.ClassBuilder.*;
+import toolbox.ModuleBuilder;
+import toolbox.ToolBox;
+
+import javadoc.tester.JavadocTester;
+
+public class TestSeeLinkAnchor extends JavadocTester {
+
+    final ToolBox tb;
+    private final Path src;
+
+    public static void main(String... args) throws Exception {
+        TestSeeLinkAnchor tester = new TestSeeLinkAnchor();
+        tester.runTests(m -> new Object[]{Paths.get(m.getName())});
+    }
+
+    TestSeeLinkAnchor() throws Exception {
+        tb = new ToolBox();
+        src = Paths.get("src");
+        generateModuleSources();
+        generatePackageSources();
+        generateInvalidLinkSource();
+        generateMissingLabelSource();
+    }
+
+    @Test
+    public void testPackage(Path base) throws Exception {
+        Path out = base.resolve("out");
+
+        javadoc("-d", out.toString(),
+                "-sourcepath", src.toString(),
+                "--no-platform-links",
+                "p1", "p2");
+
+        checkExit(Exit.OK);
+        checkOrder("p1/Class1.html",
+            """
+                    Link to <a href="../p2/package-summary.html#package-p2-heading"><code>heading in package p2</code></a>""",
+            """
+                    Plain link to <a href="../p2/Class2.html#class2-sub-heading">sub heading above</a></div>""",
+            """
+                    <li><a href="../p2/Class2.html#class2main"><code>See main heading in p2.Class2</code></a></li>
+                    <li><a href="../p2/package-summary.html#package-p2-heading"><code>See heading in p2</code></a></li>
+                    """);
+        checkOrder("p2/Class2.html",
+            """
+                    Link to <a href="#class2main"><code>local anchor</code></a>""",
+            """
+                    Plain link <a href="../p1/Class1.html#main">to Class1</a>.""");
+        checkOrder("p2/package-summary.html",
+            """
+                    <a href="Class2.html#class2-sub-heading"><code>See sub heading in p2.Class2</code></a>""");
+
+        checkOrder("p2/doc-files/file.html",
+            """
+                    Plain link to <a href="../../p1/Class1.html#main">heading in p1.ClassA</a>.""",
+            """
+                    <a href="../Class2.html#class2main"><code>See main heading in p2.ClassB</code></a>""");
+    }
+
+    @Test
+    public void testModule(Path base) throws Exception {
+        Path out = base.resolve("out");
+
+        javadoc("-d", out.toString(),
+                "--module-source-path", src.toString(),
+                "--no-platform-links",
+                "--module", "m1,m2",
+                "m2/com.m2");
+
+        checkExit(Exit.OK);
+        checkOrder("m1/module-summary.html",
+            """
+                    <a href="../m2/com/m2/Class2.html#main-heading"><code>See main heading in Class2</code></a>""");
```

Comment:
It seems like an anti-pattern to check the order of a single item!
Is this as intended?  Other examples, below, provide pairs.

Reply:
Admittedly it was not a conscious choice to use `checkOrder` here, but I don't think it is an anti-pattern. Using `checkOrder` for a single string adds no overhead compared to `checkOutput`, and If we ever add additional string I think it is ok to do an ordered check by default.

Reply:
Looking at the `JavadocTester` source code, I notice that the remaining uses of `checkOutput` are numbered and marked as TODO, which supports the idea that `checkOrder` should be the default.